os/arch/arm/src/imxrt/imxrt_flash.h

```diff
@@ -136,6 +136,7 @@ extern "C" {
 #define EXTERN extern
 #endif
 
+#if !defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
```

Comment:
If all of 105x allows only hyperflash, and 102x allows only nor, then this is acceptable, but if there's chance that use another flash on 102x or 105x, then you should make config for flash itself.
Assume that you made configs for these flashes as a below.
CONFIG_XXXX_HYPER_FLASH_YYYY... / CONFIG_XXXX_NOR_FLASH_YYYYY...

Then you can enable it in each of defconfig, also use should enable it if they want make their own .config without defconfig